src/zenml/integrations/azure/secrets_managers/azure_secrets_manager.py

```diff
@@ -139,59 +296,82 @@ def update_secret(self, secret: BaseSecretSchema) -> None:
 
         Args:
             secret: the secret to update
+
+        Raises:
+            KeyError: if the secret does not exist
         """
+        self.validate_secret_name(secret.name)
+        self._create_or_update_secret(secret)
```

Comment:
Unless my GitHub UI is bugging here, this call is duplicated from below and should probably not be called ahead of the following checks?